src/aind_metadata_mapper/core.py

```diff
@@ -2,19 +2,44 @@
 
 import argparse
 import logging
+import json
 from abc import ABC, abstractmethod
 from os import PathLike
 from pathlib import Path
-from typing import Any, Generic, Optional, TypeVar, Union
+from typing import Any, Generic, Optional, TypeVar, Union, Type
 
 from aind_data_schema.base import AindCoreModel
-from pydantic import ValidationError
+from pydantic import ValidationError, BaseModel, ConfigDict, Field
 from pydantic_settings import BaseSettings
-from aind_metadata_mapper.models import JobResponse
+
 
 _T = TypeVar("_T", bound=BaseSettings)
 
 
+class JobResponse(BaseModel):
+    """Standard model of a JobResponse."""
+
+    model_config = ConfigDict(extra="forbid")
+    status_code: int
+    message: Optional[str] = Field(None)
+    data: Optional[str] = Field(None)
+
+
+class BaseJobSettings(BaseSettings):
+    """Parent class for generating settings from a config file."""
+
+    @classmethod
+    def from_config_file(cls: Type[_T], config_path: Path) -> _T:
```

Comment:
You should use the BaseSettings built-in classmethod `settings_customise_sources` and use a config that a user can set as a field. You can see an example here: https://github.com/AllenNeuralDynamics/aind-codeocean-api/blob/main/src/aind_codeocean_api/credentials.py